### PR TITLE
closing #273: filters for public surveys

### PR DESCRIPTION
**Pull request by adam-jachymek**

### CODE REVIEW

File: team-fingerprints-web/src/routes/ProfileDetails/ProfileDetails.tsx [hunk 1]
```diff
@@ -46,6 +46,8 @@ const ProfileDetails = () => {
     }
   );
 
+  console.log("profile", data);
```
Comment (tobob) [inline team-fingerprints-web/src/routes/ProfileDetails/ProfileDetails.tsx:49]: ✂️

File: team-fingerprints-web/src/routes/ProfileDetails/ProfileDetails.tsx [hunk 2]
```diff
@@ -56,6 +58,8 @@ const ProfileDetails = () => {
     {} as FormData
   );
 
+  console.log("initialValues", initialValues);
```
Comment (tobob) [inline team-fingerprints-web/src/routes/ProfileDetails/ProfileDetails.tsx:61]: ✂️

**Review by tobob — APPROVED** [review]

**Comment by tobob** [comment]
please remove some console logs

**Review by tobob — APPROVED** [review]

File: team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/ResultsFilters.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { useQuery } from "react-query";
```
Comment (polakowski) [inline team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/ResultsFilters.tsx:1]: Please sort imports

File: team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/FiltersSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { Select } from "@mantine/core";
```
Comment (polakowski) [inline team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/FiltersSelect.tsx:1]: Please sort imports

File: team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/FiltersSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import { FilterSelect } from "../../../../types/models";
+import { useCallback, useMemo, memo } from "react";
+
+type Props = {
+  filter: FilterSelect;
+  values: any;
+  handleChange: any;
```
Comment (polakowski) [inline team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/FiltersSelect.tsx:8]: Type?

File: team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/FiltersSelect.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  handleSubmit: any;
```
Comment (polakowski) [inline team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/FiltersSelect.tsx:9]: Type?

File: team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/ResultsFilters.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+import { isEmpty, times, reduce, omitBy } from "lodash";
+import { useFormik } from "formik";
+import FiltersSelect from "./FiltersSelect";
+import { FilterSelect } from "../../../../types/models";
+import { Loader } from "@mantine/core";
+import { Filter } from "../../../../types/models";
+import "./styles.sass";
+
+type Props = {
+  setFilterValues: any;
```
Comment (polakowski) [inline team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/ResultsFilters.tsx:12]: Type?

**Review by polakowski — CHANGES_REQUESTED** [review]

File: team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/ResultsFilters.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    any,
+    Error
+  >(["surveyFilters"], async () => {
+    const { data } = await axios.get<any>(`/survey-results/companies/filters`);
+    return data;
+  });
+
+  const initialValues = {};
+
+  const { handleSubmit, handleChange, values } = useFormik({
```
[the remark this answers is not in the document]
Comment (adam-jachymek) [reply team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/ResultsFilters.tsx:26]: After adding context we will still store the same data twice (filter values in both formik and context)

[the remark this answers is not in the document]
Comment (polakowski) [reply team-fingerprints-web/src/routes/PublicResponses/Edit/resultsFilters/ResultsFilters.tsx]: Let's use formik in context so it's in one place

**Comment by adam-jachymek** [comment]
merged by bob :D

**Pull request merged**